Projects/LinkedListIterator/LinkedListIterator/LinkedListIterator/DLList.h

```diff
@@ -0,0 +1,398 @@
+	}
+
+	//-------------------------------------------------------------------------------------------
+	const T& GetHead() const
+	{
+		return Head->Data;
+	}
+
+	//-------------------------------------------------------------------------------------------
+	T& GetTail()
+	{
+		return Tail->Data;
+	}
+
+	//-------------------------------------------------------------------------------------------
+	const T& GetTail() const
+	{
+		return Tail->Data;
+	}
+
+	//-------------------------------------------------------------------------------------------
+	//											SIZE
+	//-------------------------------------------------------------------------------------------
+	bool IsEmpty() const
+	{
+		return Size == 0;
+	}
+
+	//-------------------------------------------------------------------------------------------
+	int GetSize() const
+	{
+		return Size;
+	}
+
+	//-------------------------------------------------------------------------------------------
+	//										MODIFIERS
+	//-------------------------------------------------------------------------------------------
+	void AddHead(const T& NewData)
+	{
+		FNode* Node = new FNode;
+		Node->Data = NewData;
+		Node->Next = Head;
+
+		if (!Head)
+		{
+			Tail = Node;
+		}
+		else
+		{
+			Head->Previous = Node;
+		}
+		Head = Node;
+		++Size;
+	}
+
+	//-------------------------------------------------------------------------------------------
+	void AddTail(const T& NewData)
+	{
+		if (Tail)
+		{
+			FNode* Node = new FNode;
+			Node->Data = NewData;
+			Node->Previous = Tail;
+			Tail->Next = Node;
+			Tail = Node;
+
+			++Size;
+		}
+		else
+		{
+			AddHead(NewData);
+		}
+	}
+
+	//-------------------------------------------------------------------------------------------
+	void Insert(const T& NewData, const int Index)
+	{
+		if (Index < 0 || Index > Size) //Bounds Checking
+		{
+			return;
+		}
+		else if (Index == 0) //Head
+		{
+			AddHead(NewData);
+		}
+		else if (Index == Size) //Tail
+		{
+			AddTail(NewData);
+		}
+		else if (Index < Size)
+		{
+			FNode* Current = GetNode(Index);
+			FNode* NewNode = new FNode;
+
+			NewNode->Data = NewData;
+			NewNode->Next = Current;
+			NewNode->Previous = Current->Previous;
+
+			Current->Previous->Next = NewNode;
+			Current->Previous = NewNode;
+
+			Size++;
+		}
+	}
+
+	//-------------------------------------------------------------------------------------------
+	void RemoveHead()
+	{
+		FNode* NewHead = Head->Next;
+		delete Head;
+		Head = NewHead;
+
+		if (Size == 1)
+		{
+			Tail = nullptr;
+		}
+		else
+		{
+			Head->Previous = nullptr;
+		}
+		--Size;
+	}
+
+	//-------------------------------------------------------------------------------------------
+	void Remove(const int Index)
+	{
+		if (Index < 0 || Index >= Size) //Bounds Checking
+		{
+			return;
+		}
+		else if (Index == 0) //Head
+		{
+			RemoveHead();
+		}
+		else
+		{
+			FNode* Current = GetNode(Index);
+			Current->Previous->Next = Current->Next;
+
+			if (Index == Size - 1)
```

Comment:
could you put `Current == Tail` it´s more readable but it's fine

(you can correct me if I'm wrong)